### PR TITLE
feat(api): add `lit` shorthand for constructing literals

### PR DESCRIPTION
**Pull request by cpcloud**
Add a shorthand for constructing literal values.

### CODE REVIEW

**Comment by kszucs** [comment]
What is the advantage over letting the user to alias it on import: `from ibis import literal as lit`?
I slightly prefer `ibis.literal()` over `ibis.lit()`.

**Comment by cpcloud** [comment]
Mostly for people coming from other systems that use `lit`. I personally think `literal` is annoying to type. Yeah, it's a duplicate, but with the import then users have to type that every time they open a repl to do analysis.

**Comment by kszucs** [comment]
Ok, it's fine from me. 

@jcrist @gforsyth opinions?

File: ibis/expr/api.py [hunk 1]
```diff
@@ -124,6 +125,7 @@
     'join',
     'least',
     'literal',
+    'lit',
```
Comment (kszucs) [inline ibis/expr/api.py:128]: Can you add a quick test for it so we don't accidentally remove it without notice?

**Comment by jcrist** [comment]
I don't love this alias either, but it seems harmless. +0 from me.

Curious what spurred this change though - did a user ask for it?

Also, when would a user manually call `ibis.literal`? I've used it in internal ibis code, but I don't think I've ever used it when actually using ibis.

**Comment by gforsyth** [comment]
I'm also a +0 -- no strong feelings here.

**Comment by cpcloud** [comment]
Alright, this probably isn't worth it because you can alias the import. Closing.

**Pull request closed without merging**